internal/docstore/mongodocstore/mongo.go

```diff
@@ -230,41 +230,40 @@ func (c *collection) Key(doc driver.Document) (interface{}, error) {
 // immutable, and may be of any type other than an array."
 const mongoIDField = "_id"
 
-// TODO(jba): use bulk RPCs.
 func (c *collection) RunActions(ctx context.Context, actions []*driver.Action, opts *driver.RunActionsOptions) driver.ActionListError {
-	if opts.Unordered {
-		return c.runActionsUnordered(ctx, actions, opts)
+	errs := make([]error, len(actions))
+	beforeGets, gets, writes, afterGets := driver.GroupActions(actions)
+	c.runGets(ctx, beforeGets, errs, opts)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() { defer wg.Done(); c.runGets(ctx, gets, errs, opts) }()
```

Comment:
I think you can just use a errCh and block under afterGets, seems clearer.

Reply:
Good idea. Done.